protocol/src/lib.rs

```diff
@@ -95,12 +95,35 @@ impl OutputId {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+pub struct Terminal(Option<String>, LineType);
+
+impl Terminal {
+    pub fn label(&self) -> Option<String> {
+        self.0.clone()
```

Comment:
For methods like this I like to return an borrowed str instead:

```rust
pub fn label(&self) -> Option<&str> {
    self.0.as_ref().map(String::as_str)
}
```

Reply:
Sweet. This is definitely one of the areas I'm a bit self-conscious about what I don't know, so appreciate the guidance. I'll make that change.